assets/js/base/components/chip/index.js

```diff
@@ -25,46 +25,79 @@ const Chip = ( {
 	onRemove = () => {},
 	disabled = false,
 	radius = 'small',
+	removeOnClick = false,
+	/* translators: Remove chip. */
+	ariaLabel = __( 'Remove', 'woo-gutenberg-products-block' ),
+	...props
 } ) => {
-	const Wrapper = element;
+	const Wrapper = removeOnClick ? 'button' : element;
 	const wrapperClassName = classNames(
 		className,
 		'wc-block-components-chip',
 		'wc-block-components-chip--radius-' + radius
 	);
+	const RemoveElement = ! removeOnClick ? 'button' : 'span';
+	const clickableElementProps = {
+		'aria-label':
+			ariaLabel || typeof text !== 'string'
+				? ariaLabel
+				: sprintf(
+						/* translators: %s text of the chip to remove. */
+						__( 'Remove "%s"', 'woo-gutenberg-products-block' ),
+						text
+				  ),
```

Comment:
Hmm, I'm not sure to understand you here. There are three posibilities:
* `arilaLabel` prop is set: in that case, we use `ariaLabel`.
* `ariaLabel` is not set:
  * If `text` is a string, it uses `Remove "%s"`.
  * If `text` is not a string, it uses `Remove`.

That's because `text` accepts any node, so it could be a string but also an element.

Reply:
> arilaLabel prop is set: in that case, we use ariaLabel.

In the code it's _always_ going to be set because the destructured prop value has a default:

```
ariaLabel = __( 'Remove', 'woo-gutenberg-products-block' ),
```

So the expression `ariaLabel || typeof text !== 'string'` will never evaluate the second condition _unless_ `ariaLabel` is explicitly set to something a non-truthy value and not `undefined` or not set. 

The problem I'm seeing might be more clear if you add some tests covering the conditions this expression uses.

Reply:
Aaaah, of course! :facepalm: I updated the logic in 273fde8cc42fe1a83467b8900cc10e7cf61f9d90. I also made it so it honors the screen reader text. I think it makes more sense to use that one than the default `text`.

> The problem I'm seeing might be more clear if you add some tests covering the conditions this expression uses.

I think it's now covered with the tests added in d0eeb266a71d498fdcdff3f4826185335e19da93.